### PR TITLE
Add Jekyll 3.3 release post

### PR DESCRIPTION
**Pull request by parkr**
@jekyll/core @jekyll/documentation @benbalter – for review.

- [x] Write post
- [ ] Generate list of contributors (once all PR's are in)
- [ ] Regenerate `history.md` and fix link so it points to v3.3 release notes specifically

/cc #5400

### CODE REVIEW

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+In an effort to make theme management a bit easier, any files you put into
+`/assets` in your theme will be read in as though they were part of the
+user's site. This means you can ship SCSS and CoffeeScript, images and
+webfonts, JSON and other data. Same rules apply here as in a Jekyll site:
+if it has YAML front matter, it will be converted and renderd. No YAML
+front matter, and it will simply be copied over like a static asset. Neat,
+huh?
+
+See our [documentation on the subject](/docs/assets#assets-directory) for
```
Comment (DirtyF) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: @parkr Didn't you mean to point to `/docs/themes/#assets` instead?

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+## A *lot* more!
+
+There are dozens of bug fixes and minor improvements to make your Jekyll
+experience better than ever. With every Jekyll release, we strive to bring
+greater stability and reliability to your everyday development workflow.
+
+As always, thanks to our many contributors who contributed countless hours
+of their free time to making this release happen:
+
+!! TODO: Add list of contributors for this release!
```
Comment (DirtyF) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: Here is the list returned by `git shortlog da0b45..HEAD -s`: 

Parker Moore, Ashwin Maroli, Frank Taillandier, Jussi Kinnula, Jordon Bedwell, David Zhang, Florian Thomas, XhmikosR, Anthony Gaudino, Pat Hawks, mertkahyaoglu, Jeff Kolesky, Jonathan Thornton, Nadjib Amar, Nicolas Hoizey, Manmeet Gill, Zlatan Vasović, Anatoliy Yastreb, vohedge, Antonio, Ben Balter, Charles Horn, Chris Finazzo, Eduardo Bouças, Edward Thomson, Gerardo, Júnior Messias, Kyle O'Brien, Mark H. Wilkinson, Mertcan Gökgöz, Michal Švácha, Mike Kasberg, Nicolas Porcel, Patrick Marsceill, Stephen Checkoway, Stuart Kent, shingo-nakanishi

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown (hunk 1 above)
Comment (parkr) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: Yes! Good catch. 😄

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 3]
```diff
@@ -0,0 +1,90 @@
+---
+layout: news_item
+title: 'Jekyll 3.3 is here with better theme support, new URL filters, and tons more'
+date: 2016-10-03 17:08:38 -0700
+author: parkr
+version: 3.3.0
+categories: [release]
+---
+
+We have tons of new features for you in our latest release of Jekyll. Three
```
Comment (benbalter) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: Personal nit, I prefer to [write from the user's perspective](http://ben.balter.com/2015/07/20/write-corporate-blog-posts-as-a-human/#write-for-users-not-for-yourself) "here's what you can use" more than our perspective "we have lots of features for you".

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+key things you might want to give a whirl:
+
+### 1. Themes can now ship static & dynamic assets in an `/assets` directory
+
+We're really stoked about this one. In Jekyll 3.2, we shipped the ability
+to use a theme that was packaged as a gem. Due to security necessities and
+ease-of-use concerns, this initial ship only included support for includes,
+layouts, and sass partials. A theme couldn't write any CSS, JavaScript, or
+content to your site.
```
Comment (benbalter) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: 1. Was it really security? I thought it was just us being iterative and getting *something* out the door to nail the basics before adding additional features.

2. I don't know what we need to spend the entire first paragraph of the post talking about what users used to *not* be able to do. Let's get to what they *can* do faster. e.g., "Back in 3.2 we shipped themes... we're adding to that today with..."

3. I like the human tone, but does the reader really care if we're stoked?

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+In an effort to make theme management a bit easier, any files you put into
+`/assets` in your theme will be read in as though they were part of the
+user's site. This means you can ship SCSS and CoffeeScript, images and
+webfonts, JSON and other data. Same rules apply here as in a Jekyll site:
```
Comment (benbalter) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: > JSON and other data

I don't know that we want to plant that seed. Themes should be about packaging *presentation* not *content* (that's what the site is for).

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+if it has YAML front matter, it will be converted and renderd. No YAML
+front matter, and it will simply be copied over like a static asset. Neat,
```
Comment (benbalter) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: Do you want to mention that this is only true if it doesn't already exist in the site?

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+huh?
+
+See our [documentation on the subject]({{ "/docs/themes/#assets" | relative_url }})
+for more info.
+
+### 2. `relative_url` and `absolute_url` filters
+
+Want a clean way to prepend the `baseurl` or `url` in your config? These
+new filters have you covered. When working locally, if you set your
+`baseurl` to match your deployment environment, say `baseurl: "/myproject"`,
```
Comment (benbalter) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: Do we want to use the opportunity to explain how `baseurl` *should* be used? I can see lots of users using this filter and setting their baseurl to `/` as a result.

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+development. If you are using the new `absolute_url` filter, or using
+`site.url` anywhere, you have probably had to create a development config
+which resets the `url` value to point to `http://localhost:4000`.
+
+No longer! When you run `jekyll serve`, Jekyll will build your site with
+the value of the `host`, `port`, and SSL-related options. This defaults to
+`url: http://localhost:4000`. When you are developing locally, `site.url`
+will yield `http://localhost:4000`.
+
+Note that this only applies when `JEKYLL_ENV` is equal to `development`. If
```
Comment (benbalter) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: What if `JEKYLL_ENV` isn't set?

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown (hunk 3 above)
Comment (admhlt) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: +1 for this, and good observation, Ben.

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown (hunk 8 above)
Comment (DirtyF) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: Isn't `JEKYLL_ENV` set to `development`by default?

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+to use a theme that was packaged as a gem. Due to security necessities and
+ease-of-use concerns, this initial ship only included support for includes,
+layouts, and sass partials. A theme couldn't write any CSS, JavaScript, or
+content to your site.
+
+In an effort to make theme management a bit easier, any files you put into
+`/assets` in your theme will be read in as though they were part of the
+user's site. This means you can ship SCSS and CoffeeScript, images and
+webfonts, JSON and other data. Same rules apply here as in a Jekyll site:
+if it has YAML front matter, it will be converted and renderd. No YAML
```
Comment (Crunch09) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: I'm not a native english speaker, but isn't it `rendered` ?

Comment (ashmaroli) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: yes @Crunch09, it should be `rendered` Nice catch!

**Comment by ashmaroli** [comment]
@parkr, can I ask you to add a section about `bundle install` being initiated *automatically* after new-site generation.? I think its best to give our users a heads-up and have them be aware of the `--skip-bundle` switch right away..

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown (hunk 3 above)
Comment (parkr) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: I can't get anything to sound good. This is the best I got:

> There are tons of great new quality-of-life features you can use in 3.3. Three key things you might want to give a whirl:

Comment (admhlt) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: That sounds fine to me! Though I might replace 'give a whirl' with just 'try'. But that's the tech writer in me talking. :)

**Comment by ashmaroli** [comment]
Jekyll 3.3.0.pre.rc1 _**does not**_ seem to be backwards compatible **to me..** 
- Minima ***requires*** to be updated to 2.0. 
- JekyllArchives ***requires*** to be updated. (#5417)
- JekyllAdmin ***requires*** to be updated. (#5451)
- and many other (unknown-to-me) plugins that use Jekyll 3.2 API

Correct me, if I'm wrong, but I think Jekyll 3.3 should actually be Jekyll 4 echoing @pathawks' [comment](https://github.com/jekyll/jekyll/issues/5417#issuecomment-250269704).

**Comment by Crunch09** [comment]
> JekyllAdmin requires to be updated. (#5451)

FWIW the change which caused this error is not part of the public API so i didn't think it would cause any issues.

That said, another option would be to revert this: https://github.com/jekyll/jekyll/issues/5451#issuecomment-251890408

File: site/_posts/2016-10-03-jekyll-3-3-0-released.markdown [hunk 10]
```diff
@@ -7,24 +7,26 @@ version: 3.3.0
...
-front matter, and it will simply be copied over like a static asset. Neat,
-huh?
+webfonts, and so on -- anything you'd consider a part of the
+*presentation*. Same rules apply here as in a Jekyll site: if it has YAML
+front matter, it will be converted and rendered. No YAML front matter, and
+it will simply be copied over like a static asset.
+
+Note that if a user has a file of the same path, the theme content will not
+be included in the site, i.e. a user's `/assets/main.scss` will be written
+instead of a theme's `/assets/main.scss`. It's
```
Comment (ashmaroli) [inline site/_posts/2016-10-03-jekyll-3-3-0-released.markdown]: what do think about L28 saying `be included in the site, i.e. a user's` `/assets/main.scss` `will be read and processed` instead?
L29 seems incomplete.. ends with a `It's`

**Comment by parkr** [comment]
@jekyllbot: merge +site

**Pull request merged**